### ISSUE TITLE
Support for PermissionsManager

### ISSUE THREAD

**Issue by marvin-bitterlich** [FabioZumbi12/Sponge-RankUpper#10]
It seems this plugin is not working with PermissionsManager.
https://forums.spongepowered.org/t/key-permissionmanager-v2-6-0-api-6-x-5-x-4-2-0-gb-fr-ru-and-more/12310

Would be nice to have the support for that.

**Comment by FabioZumbi12** [comment]
But what problens or errors are you having? Please, give more information!

**Comment by marvin-bitterlich** [comment]
@FabioZumbi12 doing a `/ru check` does result in an empty group. Like "you have time xy and are in group ."  I'll post a screenshot in a sec

**Comment by marvin-bitterlich** [comment]
Here is a screenshot:
![image](https://cloud.githubusercontent.com/assets/644950/22901218/3c4d7a68-f231-11e6-8f9a-156dccf5c3d8.png)
Also no promotions are happening at all. I checked the config and the groups people are in. Also no errors on startup. 

```
[13:04:38] [Server thread/INFO]: RankUpper: [All configurations loaded!]
[13:04:38] [Server thread/INFO]: RankUpper: [Language file loaded - Using: EN-US]
[13:04:38] [Server thread/INFO]: RankUpper: [Updating player times every 1 minute(s)!]
[13:04:38] [Server thread/INFO]: RankUpper: [Saving database every 5 minute(s)!]
[13:04:38] [Server thread/INFO]: RankUpper: [RankUpper enabled.]

```

**Comment by FabioZumbi12** [comment]
HUmm... pls, paste your config here too.

**Comment by marvin-bitterlich** [comment]
Sure: https://gist.github.com/Kostronor/7154bab8f897ef27b47f9790377ce678
guest is the default group and not shown and not promoted.

**Comment by djxy** [comment]
@Kostronor is it possible to have your configuration of PermissionManager?

**Comment by marvin-bitterlich** [comment]
[permissionmanager.zip](https://github.com/FabioZumbi12/Sponge-RankUpper/files/773516/permissionmanager.zip)
Sure thing @djxy I removed all users (for their privacy) but the settings should work out of the box.

**Comment by ProsperCraft** [comment]
Rankup quit working on our server one day(month ago?), i think it may have been a sponge update that killed it, thought I reported it I will have to dig though my reports.

**Comment by ProsperCraft** [comment]
We rankup based on creds, not time played.

/ru check gives
https://gyazo.com/ca7757eca13586e43bb8614f41074ca2

Configs- 
[pluginconfigs.zip](https://github.com/FabioZumbi12/Sponge-RankUpper/files/774207/pluginconfigs.zip)

**Comment by Wissi** [comment]
AWESOME Plugin. Any update @djxy? Just wondering...

We love your plugin and are waiting on the server patiently. Thanks for your great job.

**Comment by djxy** [comment]
The new version of PermissionManager will be released soon. I will check that after.

**Comment by Wissi** [comment]
@FabioZumbi12 @djxy Hi guys, since the new version of permission manager is out and fix for rank up avail or in the workings?
We'd love to have your plugin working again. Thanks,

**Comment by djxy** [comment]
You should test it, but PermissionManager respect the guidelines of Sponge for the permission service. It shouldn't have a problem.

**Comment by FabioZumbi12** [comment]
For players with single groups is working fine. I havent tested with me with more than one group. What i can tell you on my testes is: 

Theres no more loop when the player rankup and no more "player 0 minutes" when the player ha exatct 24h played (RU fix);

You can try the new version on Ore plugin and on jenkins: http://158.69.121.149:8080/job/RankUpper/

**Comment by Wissi** [comment]
@FabioZumbi12 Thanks, we have been trying the RankUpper-3.0.0-b196.jar from your Jenkins and can confirm the Time is now being reported correctly. 
But there are still some issues that are not working:
1.- Most players stay at a "Guest" (default) group and maybe 1 or 2 of 140 are being moved to a next level. The correct permission manager commands are being used.
2.- Config file is being re-ordered. We delete the re-ordered file and upload a correct file and it too will get re-ordered. Not sure this is part of the issue.
3.- When doing:``` /ru check (<player> or self) ```it reports group none. Does not read from the permission manager Group they are at(?) not sure if flat-file should have this info but it doesn't.

![group none](https://cloud.githubusercontent.com/assets/23312242/24335255/451c39a8-123f-11e7-800e-8fdfe7005d17.png)

(uploading config files in .zip and .rar on 2 different browsers and getting this msg: We don’t support that file type.  Try again with a PNG, GIF, JPG, DOCX, PPTX, XLSX, TXT, PDF, or ZIP)

Original File:
```
# Stop counting time when a player is AFK? (Using Nucleus' API only!)
# Setting this true without Nucleus (0.23.1+) installed will result in player's time not being counted.
afk-support=false
# Date format to save data info of players.
date-format="dd/MM/yyyy"
# Enable debug messages?
debug-messages=false
# List of groups which will not be considered when checking for possible rank up scenarios.
exclude-groups=[
    "Admin group",
    "Dev group",
    "Mod group"
]
# Save to file every X minutes.
flat-file-save-interval=5
# Date format to save data info of players.
language=EN-US
# All configurations for promote players based on requeriments.
ranked-groups {
    # ##################################################################################
    # IMPORTANT: Change from "Guest" to exact group name the player need to be in to be promoted to nex group.
    # Use the placeholder {oldgroup} on commands to remove player from this group to put on next group.
    # ##################################################################################
    Guest {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 100"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=120
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Newbie
    }
    Newbie {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 200"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=240
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Rookie
    }    
    Rookie {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 300"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=480
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Trainee
    }
    Trainee {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 400"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=960
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Member
    }
    Member {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 100"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=1800
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Citizen
    }
    Citizen {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 600"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=2700
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Colonist
    }
    Colonist {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 700"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=4500
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Mayor
    }
    Mayor {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 800"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=7800
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Governor
    }
    Governor {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 900"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=10800
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Veteran
    }
    Veteran {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 1000"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=15000
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Royal
    }
}
# Interval to add for online players.
update-player-time-minutes=3
# Use uuids to store players stats on playerstats.conf?
use-uuids-instead-names=true
```
Alphabetically re-ordered File on Server:
```
# Stop counting time when a player is AFK? (Using Nucleus' API only!)
# Setting this true without Nucleus (0.23.1+) installed will result in player's time not being counted.
afk-support=false
# Date format to save data info of players.
date-format="dd/MM/yyyy"
# Enable debug messages?
debug-messages=false
# List of groups which will not be considered when checking for possible rank up scenarios.
exclude-groups=[
    "Admin group",
    "Dev group",
    "Mod group"
]
# Save to file every X minutes.
flat-file-save-interval=5
# Date format to save data info of players.
language=EN-US
# All configurations for promote players based on requeriments.
ranked-groups {
    Citizen {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 600"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=2700
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Colonist
    }
    Colonist {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 700"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=4500
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Mayor
    }
    Governor {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 900"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=10800
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Veteran
    }
    # ##################################################################################
    # IMPORTANT: Change from "Guest" to exact group name the player need to be in to be promoted to nex group.
    # Use the placeholder {oldgroup} on commands to remove player from this group to put on next group.
    # ##################################################################################
    Guest {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 100"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=120
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Newbie
    }
    Mayor {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 800"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=7800
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Governor
    }
    Member {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 100"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=1800
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Citizen
    }
    Newbie {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 200"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=240
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Rookie
    }
    Rookie {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 300"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=480
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Trainee
    }
    Trainee {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 400"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=960
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Member
    }
    Veteran {
        # Commands to execute when promote. These commands will depend on your permission plugin.
        execute-commands=[
            "pm users {player} add group {newgroup}",
            "pm users {player} remove group {oldgroup}",
            "pm save users {player}",
            "vpay Server {player} 1000"
        ]
        # Levels(not experience) needed to promote.
        levels-needed=0
        # Broadcast the promote messsage to all players.
        message-broadcast="&a>> The player &6{player} &ahas played for &6{time} &aand now is rank {newgroup} of server."
        # Minutes played needed.
        minutes-needed=15000
        # Money needed. Do not requires additional plugin.
        money-needed=0
        next-group=Royal
    }
}
# Interval to add for online players.
update-player-time-minutes=3
# Use uuids to store players stats on playerstats.conf?
use-uuids-instead-names=true
```

**Comment by Raycoms** [comment]
@FabioZumbi12 are you still working on this?
We're trying this with the newest forge and sponge versions and it does not rank up the players accordingly.

**Comment by FabioZumbi12** [comment]
I updated using LuckPerms and on all MY tests works fine. Download form ORE or from jenkins to test.

**Comment by Raycoms** [comment]
Yeah it works with LuckPerms but not with PermissionManager

2017-08-23 11:11 GMT+01:00 FabioZumbi12 <notifications@github.com>:

**Comment by Wissi** [comment]
Well, shouldn't close the Ticket as it's not working with PermissionManager (which is the title of the Ticket.).
Tried latest Jenkins, still not working with PermissionManager.

**Comment by FabioZumbi12** [comment]
Its because i will not support specific plugins. The way i build on latest build will get the groups, including inherits like the permissions API/plugins do in order of groups. I dont know but i in on PermissionManager Discord and seems is not updated with latest API updates. Maybe you should migrate to LuckPerms.
